src/Domain/OpcUAConnector.java

```diff
@@ -0,0 +1,40 @@
+package Domain;
+
+import org.apache.log4j.varia.NullAppender;
+import org.eclipse.milo.opcua.sdk.client.OpcUaClient;
+import org.eclipse.milo.opcua.sdk.client.api.config.OpcUaClientConfigBuilder;
+import org.eclipse.milo.opcua.stack.client.DiscoveryClient;
+import org.eclipse.milo.opcua.stack.core.types.structured.EndpointDescription;
+
+import java.util.List;
+
+public class OpcUAConnector {
+
+    public static OpcUaClient client = null;
+
+    public static OpcUaClient getOpcUaClient() {
+
+
+        try {
+            org.apache.log4j.BasicConfigurator.configure(new NullAppender());
+            List<EndpointDescription> endpoints = DiscoveryClient.getEndpoints("opc.tcp://127.0.0.1:4840").get();
+            //System.out.println("Endpoints - " + endpoints);
+
+            /*now we want to read specific nodes from the simulator
+            firstly we create a OpcUaClientConfigBuilder object */
+            OpcUaClientConfigBuilder configBuilder = new OpcUaClientConfigBuilder();
+            // we set the endpoint of our object to the first endpoint from the list
+            configBuilder.setEndpoint(endpoints.get(0));
+
+            /*now we want to create a OpcUaClient object with the configuration from the previously created
+            config object*/
+            client = OpcUaClient.create(configBuilder.build());
+            client.connect().get();
+        } catch (Exception e) {
+            System.out.println(e.getMessage());
+        }
+
+        return client;
+
+    }
+}
```

Comment:
Not sure what this class is supposed to do, we already have a read class that includes this. And it seems that the machine control class is supposed to handle write actions to the machine. So if it is only to connect without any other purpose, i suggest we delete this